File: fix_array_commas.py
#!/usr/bin/env python3
"""
Script to fix missing commas between array elements.
"""

import os
import re
import glob

def fix_array_commas(file_path):
    """Fix missing commas between array elements."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix missing commas between objects in arrays
        # Pattern: }\n    {
        content = re.sub(r'}\s*\n\s*{', '},\n    {', content)
        
        # Fix missing commas after closing braces in arrays
        content = re.sub(r'}\s*\n\s*(\w+)', r'},\n    \1', content)
        
        # Fix missing commas after closing braces followed by other elements
        content = re.sub(r'}\s*\n\s*(\w+:)', r'},\n    \1', content)
        
        # Write back if changes were made
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed array commas in: {file_path}")
            return True
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix array commas."""
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'components/**/*.tsx',
        'components/**/*.ts'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            # Skip .original files
            if '.original' in file_path:
                continue
                
            files_processed += 1
            if fix_array_commas(file_path):
                files_fixed += 1
    
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed array commas in {files_fixed} files")

if __name__ == "__main__":
    main()

File: fix-array-commas.js
const fs = require('fs');
const path = require('path');

function fixArrayCommas(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Fix missing commas in array elements
    // Pattern: { name: "value" } { name: "value" } -> { name: "value" }, { name: "value" }
    content = content.replace(/\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}/g, 
      `{ name: "$1", path: "$2" },
                { name: "$3", path: "$4" },
                { name: "$5", path: "$6" },
                { name: "$7", path: "$8" }`);
    
    // Fix missing commas in benefits arrays
    content = content.replace(/'([^']+)'\s*'([^']+)'\s*'([^']+)'\s*'([^']+)'/g, 
      `"$1",
      "$2",
      "$3",
      "$4"`);
    
    // Fix missing commas in features arrays
    content = content.replace(/'([^']+)'\s*'([^']+)'\s*'([^']+)'\s*'([^']+)'/g, 
      `"$1",
      "$2",
      "$3",
      "$4"`);
    
    // Fix missing commas in object properties
    content = content.replace(/(\w+:\s*"[^"]+")\s*\n\s*(\w+:\s*"[^"]+")/g, '$1,\n    $2');
    
    // Fix specific patterns
    content = content.replace(/\{\s*name:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*\}/g, 
      `{ name: "$1" },
                { name: "$2" },
                { name: "$3" },
                { name: "$4" }`);
    
    // Fix array syntax in metadata
    content = content.replace(/\[\s*\{\s*name:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*\}\s*\]/g, 
      `[
                { name: "$1" },
                { name: "$2" },
                { name: "$3" },
                { name: "$4" }
              ]`);
    
    // Fix missing commas in object literals
    content = content.replace(/(\w+:\s*"[^"]+")\s*\n\s*(\w+:\s*"[^"]+")/g, '$1,\n    $2');
    
    // Fix specific syntax issues
    content = content.replace(/\{\s*name:\s*"([^"]+)"\s*\}\s*\{/g, '{ name: "$1" }, {');
    
    // Fix array syntax issues
    content = content.replace(/\[\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\{\s*name:\s*"([^"]+)"\s*path:\s*"([^"]+)"\s*\}\s*\]/g, 
      `[
                { name: "$1", path: "$2" },
                { name: "$3", path: "$4" },
                { name: "$5", path: "$6" },
                { name: "$7", path: "$8" }
              ]`);
    
    fs.writeFileSync(filePath, content);
    console.log(`Fixed: ${filePath}`);
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
  }
}

function walkDirectory(dir) {
  const files = fs.readdirSync(dir);
  
  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      walkDirectory(filePath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      fixArrayCommas(filePath);
    }
  }
}

// Start fixing from the app directory
walkDirectory('./app');
console.log('Array comma fixing completed!');